### PR TITLE
Kafka instrumentation part 2 - Integration test

### PR DESCRIPTION
**Pull request by pellared**
## Why

Prepare a failing test using the sample application.

## What

- Add `KafkaTests`
- Add timeout in `TestHelper.RunSampleAndWaitForExit`
- Update package versions for test matrix
- Fix `Samples.AspNetMvc5`
- Remove unexisting samples from sln

## Testing

```
rpajak@LAPTOP-K6FOL8LO MINGW64 ~/repos/signalfx-dotnet-tracing/test/Datadog.Trace.ClrProfiler.IntegrationTests (kafka-part-2)
$ dotnet test --filter KafkaTests
  Determining projects to restore...
  All projects are up-to-date for restore.
  Datadog.Trace.ClrProfiler.Managed.Core -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed.Core\bin\Debug\netstandard2.0\SignalFx.Tracing.ClrProfiler.Managed.Core.dll
  Datadog.Core.Tools -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\tools\Datadog.Core.Tools\bin\Debug\net461\Datadog.Core.Tools.dll
  Datadog.Trace -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace\bin\Debug\netstandard2.0\SignalFx.Tracing.dll
  Datadog.Trace -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace\bin\Debug\net461\SignalFx.Tracing.dll
  Datadog.Trace.ClrProfiler.Managed.Core -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed.Core\bin\Debug\net45\SignalFx.Tracing.ClrProfiler.Managed.Core.dll
  Datadog.Trace.ClrProfiler.Managed.Core -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed.Core\bin\Debug\net461\SignalFx.Tracing.ClrProfiler.Managed.Core.dll
  Datadog.Trace -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace\bin\Debug\net45\SignalFx.Tracing.dll
  Datadog.Trace.OpenTracing -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.OpenTracing\bin\Debug\netstandard2.0\SignalFx.Tracing.OpenTracing.dll
  Datadog.Trace.OpenTracing -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.OpenTracing\bin\Debug\net461\SignalFx.Tracing.OpenTracing.dll
  Datadog.Trace.OpenTracing -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.OpenTracing\bin\Debug\net45\SignalFx.Tracing.OpenTracing.dll
  Datadog.Trace.AspNet -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.AspNet\bin\Debug\net45\SignalFx.Tracing.AspNet.dll
  Datadog.Trace.ClrProfiler.Managed -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed\bin\Debug\netstandard2.0\SignalFx.Tracing.ClrProfiler.Managed.dll
  Datadog.Trace.TestHelpers -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.TestHelpers\bin\Debug\net461\Datadog.Trace.TestHelpers.dll
  Datadog.Core.Tools -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\tools\Datadog.Core.Tools\bin\Debug\netcoreapp3.1\Datadog.Core.Tools.dll
  Datadog.Core.Tools -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\tools\Datadog.Core.Tools\bin\Debug\netcoreapp2.1\Datadog.Core.Tools.dll
  Datadog.Trace.ClrProfiler.Managed -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed\bin\Debug\net461\SignalFx.Tracing.ClrProfiler.Managed.dll
  Datadog.Trace.ClrProfiler.Managed -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Managed\bin\Debug\net45\SignalFx.Tracing.ClrProfiler.Managed.dll
  Datadog.Trace.TestHelpers -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.TestHelpers\bin\Debug\netcoreapp3.0\Datadog.Trace.TestHelpers.dll
  Datadog.Trace.TestHelpers -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.TestHelpers\bin\Debug\netcoreapp2.1\Datadog.Trace.TestHelpers.dll
  Datadog.Core.Tools -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\tools\Datadog.Core.Tools\bin\Debug\net452\Datadog.Core.Tools.dll
  Datadog.Trace.ClrProfiler.IntegrationTests -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp2.1\Datadog.Trace.ClrProfiler.IntegrationTests.dll
  Datadog.Trace.TestHelpers -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.TestHelpers\bin\Debug\netcoreapp3.1\Datadog.Trace.TestHelpers.dll
  Datadog.Trace.ClrProfiler.IntegrationTests -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\net461\Datadog.Trace.ClrProfiler.IntegrationTests.dll
  Datadog.Trace.ClrProfiler.IntegrationTests -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp3.0\Datadog.Trace.ClrProfiler.IntegrationTests.dll
  Datadog.Trace.TestHelpers -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.TestHelpers\bin\Debug\net452\Datadog.Trace.TestHelpers.dll
  Datadog.Trace.ClrProfiler.IntegrationTests -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp3.1\Datadog.Trace.ClrProfiler.IntegrationTests.dll
  Datadog.Trace.ClrProfiler.IntegrationTests -> C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\net452\Datadog.Trace.ClrProfiler.IntegrationTests.dll
Test run for C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\net452\Datadog.Trace.ClrProfiler.IntegrationTests.dll (.NETFramework,Version=v4.5.2)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.9105678]     Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [FAIL]
  Failed Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [256 ms]
  Error Message:
   System.Exception : application not found: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net452\Samples.Kafka.exe
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.TestHelper.StartSample(Int32 traceAgentPort, String arguments, String packageVersion, Int32 aspNetCorePort, Dictionary`2 envVars) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\Helpers\TestHelper.cs:line 74
   at Datadog.Trace.ClrProfiler.IntegrationTests.TestHelper.RunSampleAndWaitForExit(Int32 traceAgentPort, String arguments, String packageVersion, Dictionary`2 envVars) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\Helpers\TestHelper.cs:line 91
   at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\KafkaTests.cs:line 23
  Standard Output Messages:
 Platform: X86
 Configuration: Debug
 TargetFramework: net452
 .NET Core: False
 Application: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net452\Samples.Kafka.exe
 Attempt 1: Unable to find profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net452\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Attempt 2: Unable to find profiler at C:\Users\rpajak\AppData\Local\Temp\99c4248f-8850-4a92-8458-d7047c79035e\99c4248f-8850-4a92-8458-d7047c79035e\assembly\dl3\6abe6a14\b1960d39_e868d701\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Found profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Native\bin\Debug\x86\SignalFx.Tracing.ClrProfiler.Native.dll.
 Profiler DLL: C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Native\bin\Debug\x86\SignalFx.Tracing.ClrProfiler.Native.dll



Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration:  - Datadog.Trace.ClrProfiler.IntegrationTests.dll (net452)
Test run for C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\net461\Datadog.Trace.ClrProfiler.IntegrationTests.dll (.NETFramework,Version=v4.6.1)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:01.4554833]     Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [FAIL]
  Failed Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [256 ms]
  Error Message:
   System.Exception : application not found: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net461\Samples.Kafka.exe
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.TestHelper.StartSample(Int32 traceAgentPort, String arguments, String packageVersion, Int32 aspNetCorePort, Dictionary`2 envVars) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\Helpers\TestHelper.cs:line 74
   at Datadog.Trace.ClrProfiler.IntegrationTests.TestHelper.RunSampleAndWaitForExit(Int32 traceAgentPort, String arguments, String packageVersion, Dictionary`2 envVars) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\Helpers\TestHelper.cs:line 91
   at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\KafkaTests.cs:line 23
  Standard Output Messages:
 Platform: X86
 Configuration: Debug
 TargetFramework: net461
 .NET Core: False
 Application: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net461\Samples.Kafka.exe
 Attempt 1: Unable to find profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X86\Debug\net461\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Attempt 2: Unable to find profiler at C:\Users\rpajak\AppData\Local\Temp\06e840d0-538b-4bea-b7fa-d13d01d8f823\06e840d0-538b-4bea-b7fa-d13d01d8f823\assembly\dl3\f181639c\9060de37_e868d701\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Found profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Native\bin\Debug\x86\SignalFx.Tracing.ClrProfiler.Native.dll.
 Profiler DLL: C:\Users\rpajak\repos\signalfx-dotnet-tracing\src\Datadog.Trace.ClrProfiler.Native\bin\Debug\x86\SignalFx.Tracing.ClrProfiler.Native.dll



Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration:  - Datadog.Trace.ClrProfiler.IntegrationTests.dll (net461)
Test run for C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp2.1\Datadog.Trace.ClrProfiler.IntegrationTests.dll (.NETCoreApp,Version=v2.1)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:07.9763482]     Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [FAIL]
  Failed Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [7 s]
  Error Message:
   Expecting at least 2 spans, but received 0
Expected: True
Actual:   False
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\KafkaTests.cs:line 29
  Standard Output Messages:
 Platform: X64
 Configuration: Debug
 TargetFramework: netcoreapp2.1
 .NET Core: True
 Application: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp2.1\Samples.Kafka.dll
 Found profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp2.1\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Profiler DLL: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp2.1\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll
 Found integrations at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp2.1\profiler-lib\integrations.json.
 StandardOutput:
 consumed: test value




Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration: < 1 ms - Datadog.Trace.ClrProfiler.IntegrationTests.dll (netcoreapp2.1)
Test run for C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp3.0\Datadog.Trace.ClrProfiler.IntegrationTests.dll (.NETCoreApp,Version=v3.0)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:07.8574219]     Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [FAIL]
  Failed Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [7 s]
  Error Message:
   Expecting at least 2 spans, but received 0
Expected: True
Actual:   False
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\KafkaTests.cs:line 29
  Standard Output Messages:
 Platform: X64
 Configuration: Debug
 TargetFramework: netcoreapp3.0
 .NET Core: True
 Application: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.0\Samples.Kafka.dll
 Found profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.0\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Profiler DLL: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.0\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll
 Found integrations at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.0\profiler-lib\integrations.json.
 StandardOutput:
 consumed: test value




Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration: < 1 ms - Datadog.Trace.ClrProfiler.IntegrationTests.dll (netcoreapp3.0)
Test run for C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\bin\Debug\netcoreapp3.1\Datadog.Trace.ClrProfiler.IntegrationTests.dll (.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:09.0458823]     Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [FAIL]
  Failed Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "") [8 s]
  Error Message:
   Expecting at least 2 spans, but received 0
Expected: True
Actual:   False
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in C:\Users\rpajak\repos\signalfx-dotnet-tracing\test\Datadog.Trace.ClrProfiler.IntegrationTests\KafkaTests.cs:line 29
  Standard Output Messages:
 Platform: X64
 Configuration: Debug
 TargetFramework: netcoreapp3.1
 .NET Core: True
 Application: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.1\Samples.Kafka.dll
 Found profiler at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.1\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll.
 Profiler DLL: C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.1\profiler-lib\SignalFx.Tracing.ClrProfiler.Native.dll
 Found integrations at C:\Users\rpajak\repos\signalfx-dotnet-tracing\samples\Samples.Kafka\bin\X64\Debug\netcoreapp3.1\profiler-lib\integrations.json.
 StandardOutput:
 consumed: test value




Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration: < 1 ms - Datadog.Trace.ClrProfiler.IntegrationTests.dll (netcoreapp3.1)
```

Azure Pipeline fails as expected:

```
  X Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(packageVersion: "1.5.3") [11s 566ms]
  Error Message:
   Expecting at least 2 spans, but received 0
Expected: True
Actual:   False
  Stack Trace:
     at Datadog.Trace.ClrProfiler.IntegrationTests.KafkaTests.SubmitsTraces(String packageVersion) in /project/test/Datadog.Trace.ClrProfiler.IntegrationTests/KafkaTests.cs:line 29
  Standard Output Messages:
 Platform: X64
 Configuration: Debug
 TargetFramework: netcoreapp3.1
 .NET Core: True
 Application: /project/samples/Samples.Kafka/bin/Debug/netcoreapp3.1/publish/Samples.Kafka.dll
 Attempt 1: Unable to find profiler at /project/samples/Samples.Kafka/bin/Debug/netcoreapp3.1/publish/profiler-lib/SignalFx.Tracing.ClrProfiler.Native.so.
 Attempt 2: Unable to find profiler at /project/test/Datadog.Trace.ClrProfiler.IntegrationTests/bin/Debug/netcoreapp3.1/publish/profiler-lib/SignalFx.Tracing.ClrProfiler.Native.so.
 Found profiler at /project/src/Datadog.Trace.ClrProfiler.Native/bin/Debug/x64/SignalFx.Tracing.ClrProfiler.Native.so.
 Profiler DLL: /project/src/Datadog.Trace.ClrProfiler.Native/bin/Debug/x64/SignalFx.Tracing.ClrProfiler.Native.so
 Found integrations at /project/samples/Samples.Kafka/bin/Debug/netcoreapp3.1/publish/profiler-lib/integrations.json.
 StandardOutput:
 consumed: test value
```

## Next planned PR

Implement instrumentation that would make the test green.

### CODE REVIEW

File: samples-aspnet/Samples.AspNetMvc5/Controllers/DistributedTracingMvcController.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public async Task<ActionResult> Distributed()
                         span.ServiceName,
                         span.OperationName,
                         span.ResourceName,
-                        span.TraceId,
+                        span.TraceId.Lower,
```
Comment (dszmigielski) [inline samples-aspnet/Samples.AspNetMvc5/Controllers/DistributedTracingMvcController.cs:42]: 🤫

**Review by dszmigielski — APPROVED** [review]
👍

**Comment by pellared** [comment]
Looks like I made some `docker compose` configuration error:

```
Timeout after 00:01:00
 StandardError:
 %3|1624554698.339|FAIL|rdkafka#producer-1| [thrd:localhost:9092/1001]: localhost:9092/1001: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)
 %3|1624554698.339|ERROR|rdkafka#producer-1| [thrd:localhost:9092/1001]: localhost:9092/1001: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)
 %3|1624554709.441|FAIL|rdkafka#consumer-2| [thrd:GroupCoordinator]: GroupCoordinator: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)
 %3|1624554709.441|ERROR|rdkafka#consumer-2| [thrd:GroupCoordinator]: GroupCoordinator: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)
```

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/TestHelper.cs [hunk 2]
```diff
@@ -86,26 +87,43 @@ public Process StartSample(int traceAgentPort, string arguments, string packageV
                 envVars: envVars);
         }
 
-        public ProcessResult RunSampleAndWaitForExit(int traceAgentPort, string arguments = null, string packageVersion = "", Dictionary<string, string> envVars = null)
+        public ProcessResult RunSampleAndWaitForExit(int traceAgentPort, string arguments = null, string packageVersion = "", Dictionary<string, string> envVars = null, TimeSpan timeout = default)
```
Comment (pjanotti) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/TestHelper.cs:90]: 👍 ❤️

**Review by pjanotti — APPROVED** [review]
Really nice!

**Pull request merged**